### PR TITLE
realize methods equals and hash

### PR DESCRIPTION
**Pull request by RostykRosolovskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +30,25 @@ public Integer getLength() {
...
+        if (this == obj) {
+            return true;
+        }
+        {
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            Rectangle rectangle = (Rectangle) obj;
+            return Objects.equals(getWidth(), rectangle.getWidth())
+                    && Objects.equals(getLength(), rectangle.getLength());
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: what about color?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +30,25 @@ public Integer getLength() {
...
+        }
+    }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java:47]: remove redundant line

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +30,25 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: do we need this {}?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +30,23 @@ public Integer getLength() {
...
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(getWidth(), rectangle.getWidth())
+                && Objects.equals(getLength(), rectangle.getLength())
+                && Objects.equals(getColor(), rectangle.getColor());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getWidth(), getLength(), getColor());
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java:50]: @RostykRosolovskyi lets not use java.util.Objects;

**Pull request closed without merging**